cfdm/read_write/netcdf/netcdfread.py

```diff
@@ -584,6 +585,66 @@ def is_cdl_file(cls, filename):
 
         return cdl
 
+    @classmethod
+    def is_file(cls, filename):
+        """Return `True` if *filename* is a file.
+
+        Note that a remote URL starting with ``http://`` or
+        ``https://`` is always considered as a file.
+
+        .. versionadded:: (cfdm) 1.10.1.1
+
+        :Parameters:
+
+            filename: `str`
+                The name of the file.
+
+        :Returns:
+
+            `bool`
+                Whether or not *filename* is a file.
+
+        **Examples**
+
+        >>> {{package}}.{{class}}.is_file('file.nc')
+        True
+        >>> {{package}}.{{class}}.is_file('http://file.nc')
+        True
+        >>> {{package}}.{{class}}.is_file('https://file.nc')
+        True
+
+        """
+        # Assume that URLs are files
+        u = urlparse(filename)
+        if u.scheme in ("http", "https"):
+            return True
+
+        return os.path.isfile(filename)
+
+    @classmethod
+    def is_dir(cls, filename):
+        """Return `False` if *filename* is a directory.
```

Comment:
```suggestion
        """Return `True` if *filename* is a directory.
```

unless we want to cause confusion left, right and centre :)